xdem/coreg.py

```diff
@@ -1063,3 +1081,620 @@ def coregister(reference_raster: Union[str, gu.georaster.Raster], to_be_aligned_
+            raise ValueError(f"Incompatible add type: {type(other)}. Expected 'Coreg' subclass")
+        return CoregPipeline([self, other])
+
+    def _fit_func(self, ref_dem: np.ndarray, tba_dem: np.ndarray, transform: Optional[rio.transform.Affine],
+                  weights: Optional[np.ndarray]):
+        raise NotImplementedError("This should have been implemented by subclassing")
+
+    def _apply_func(self, dem: np.ndarray, transform: rio.transform.Affine) -> np.ndarray:
+        raise NotImplementedError("This should have been implemented by subclassing")
+
+    def _apply_pts_func(self, coords: np.ndarray) -> np.ndarray:
+        raise NotImplementedError("This should have been implemented by subclassing")
+
+    def _to_matrix_func(self) -> np.ndarray:
+        raise NotImplementedError("This should be implemented by subclassing")
+
+
+class BiasCorr(Coreg):
+    """
+    DEM bias correction.
+
+    Estimates the mean (or median, weighted avg., etc.) offset between two DEMs.
+    """
+
+    def __init__(self, bias_func=np.average):  # pylint: disable=super-init-not-called
+        """
+        Instantiate a bias correction object.
+
+        :param bias_func: The function to use for calculating the bias. Default: (weighted) average.
+        """
+        self._meta: dict[str, Any] = {"bias_func": bias_func}
+
+    def _fit_func(self, ref_dem: np.ndarray, tba_dem: np.ndarray, transform: Optional[rio.transform.Affine],
+                  weights: Optional[np.ndarray]):
+        """Estimate the bias using the bias_func."""
+        diff = ref_dem - tba_dem
+        diff = diff[np.isfinite(diff)]
+
+        # Use weights if those were provided.
+        bias = self._meta["bias_func"](diff) if weights is None \
+            else self._meta["bias_func"](diff, weights=weights)
+
+        self._meta["bias"] = bias
+
+    def _apply_func(self, dem: np.ndarray, transform: rio.transform.Affine) -> np.ndarray:
+        """Apply the bias to a DEM."""
+        return dem + self._meta["bias"]
+
+    def _apply_pts_func(self, coords: np.ndarray):
+        """Apply the bias to a given coordinate array."""
+        new_coords = coords.copy()
+        new_coords[:, 2] += self._apply_func(coords[:, 2], None)  # type: ignore
+
+        return new_coords
+
+    def _to_matrix_func(self) -> np.ndarray:
+        """Convert the bias to a transform matrix."""
+        empty_matrix = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]], dtype=float)
+
+        empty_matrix[2, 3] += self._meta["bias"]
+
+        return empty_matrix
+
+
+class ICP(Coreg):
+    """
+    Iterative Closest Point DEM coregistration.
+
+    Estimates a rigid transform (rotation + translation) between two DEMs.
+
+    Requires 'opencv'
+    """
+
+    def __init__(self, max_iterations=100, tolerance=0.05, rejection_scale=2.5, num_levels=6):  # pylint: disable=super-init-not-called
+        """
+        Instantiate an ICP coregistration object.
```

Comment:
Maybe put a link to the PDAL ICP documentation for details about the meaning of the parameters?

Reply:
Well, opencv's documentation, I forgot you use OpenCV's ICP now!

Reply:
Added in 56a4d1b